packages/toolpad-app/src/server/appTemplateDoms/default.json

```diff
@@ -13,7 +13,7 @@
           "value": {
             "url": {
               "type": "const",
-              "value": "http://localhost:3000/static/employees.json"
+              "value": "https://demo.toolpad.io/static/employees.json"
```

Comment:
could we do it like the movies are doing it?
```
export const MOVIES_API_DEMO_URL = new URL('/static/movies.json', config.externalUrl).href;
```

Reply:
ah we can't, this works for now.

Reply:
The issue is that it needs to live inside the `default.json` file, so we would need to do some sort of string-replace at build-time?

Reply:
Eventually we should either generate the template from `config.externalUrl`, or serve these files from somewhere that we can access from anywhere.

Reply:
> or serve these files from somewhere that we can access from anywhere.

`https://demo.toolpad.io/static` serves this purpose?

Reply:
Once we have our own definitive cloud domain I guess that should be fine, so until then demo should work.

Reply:
This fix doesn't work for local deployments because of a CORS issue for on the remote static file. The other option I could think of is to create two separate versions of the demo dom, one pointing to local and one to `https://demo.toolpad.io`, but that wouldn't work on our `master--toolpad.mui.com` deployments.

Thinking of an alternative fix

Reply:
This can be done by writing a migration (`v4.ts`) that applies that string replacement on the default dom at build time. Should work, I'll also add a test to make sure this works.

Reply:
I guess we could just use `require` for dynamic templates instead of `readJsonFile`. Let me write a suggestion maybe.

Reply:
Something like this? Would probably need some cleanup and adjusting, I can write a PR too:

```
import * as path from 'path';
import * as appDom from '../../appDom';
import { migrateUp } from '../../appDom/migrations';
import { AppTemplateId } from '../../types';
import { readJsonFile } from '../../utils/fs';
import config from '../config';
import projectRoot from '../projectRoot';

const DOMS_DIR_PATH = './src/server/appTemplateDoms';

const APP_TEMPLATE_DOM_PATHS: Record<AppTemplateId, string | null> = {
  blank: {
    path: null,
    isDynamic: false,
  ,
  images: {
    path: path.resolve(
      projectRoot,
      DOMS_DIR_PATH,
      // @TODO: Remove demo template once demo supports server-side queries
      config.isDemo ? './images-demo.json' : './images.json',
    ),
    isDynamic: false,
  }
  default: {
    path: path.resolve(projectRoot, DOMS_DIR_PATH, './default.json'),
    isDynamic: true,
  }
};

export async function getAppTemplateDom(
  appTemplateId: AppTemplateId,
): Promise<appDom.AppDom | null> {
  const appTemplateDom = APP_TEMPLATE_DOM_PATHS[appTemplateId];
  if (!appTemplateDom) return null
  
  const dom = appTemplateDom.isDynamic ? require(appTemplateDom.path) : await readJsonFile(appTemplateDom.path);
  return dom ? migrateUp(dom) : null;
}

Reply:
> This makes sense longer term though. For now, I think the fix lies in writing a migration since that gives us a fix for all apps that have already been created as a free side-effect

Sounds good if we need to fix the already created apps.
I'll make a PR with dynamic templates, shouldn't be too difficult and then we can use better solutions in the future.

Reply:
Here's my idea, seems to work https://github.com/mui/mui-toolpad/pull/1430